docs/how-to-guides/create-a-user-defined-function.md

```diff
@@ -0,0 +1,913 @@
+        public Struct initialize() {
+            return new Struct(AGGREGATE_SCHEMA);
+        }
+
+        @Override
+        public Struct aggregate(Struct newValue, Struct aggregateValue) {
+            long c = newValue.getInt64("C");
+            
+            long min = Math.min(c, getMin(aggregateValue));
+            long max = Math.max(c, getMax(aggregateValue));
+            long count = (getCount(aggregateValue) + 1);
+
+            aggregateValue.put("MIN", min);
+            aggregateValue.put("MAX", max);
+            aggregateValue.put("COUNT", count);
+            
+            return aggregateValue;
+        }
+
+        @Override
+        public Struct map(Struct intermediate) {
+            Struct result = new Struct(RETURN_SCHEMA);
+
+            long min = intermediate.getInt64("MIN");
+            long max = intermediate.getInt64("MAX");
+
+            result.put("MIN", min);
+            result.put("MAX", max);
+            result.put("COUNT", intermediate.getInt64("COUNT"));
+            result.put("DIFFERENTIAL", max - min);
+
+            return result;
+        }
+
+        @Override
+        public Struct merge(Struct aggOne, Struct aggTwo) {
+            return aggOne;
+        }
+
+        private Long getMin(Struct aggregateValue) {
+            Long result = aggregateValue.getInt64("MIN");
+
+            if (result != null) {
+                return result;
+            } else {
+                return Long.MAX_VALUE;
+            }
+        }
+
+        private Long getMax(Struct aggregateValue) {
+            Long result = aggregateValue.getInt64("MAX");
+
+            if (result != null) {
+                return result;
+            } else {
+                return Long.MIN_VALUE;
+            }
+        }
+
+        private Long getCount(Struct aggregateValue) {
+            Long result = aggregateValue.getInt64("COUNT");
+
+            if (result != null) {
+                return result;
+            } else {
+                return 0L;
+            }
+        }
+    }
+}
+```
+
+There are a few important things to call out in this class:
+
+- Schemas are declared for the input struct parameter, intermediate aggregation struct value, and output struct value. Becasue each of these three structs are different, they need their own schemas.
+
+- Descriptor strings are created for each schema, too. This communicates the underlying types to ksqlDB is a way that its type system can understand. In the future, this may become automated, but it must be maintained today.
+
+- The schemas, and all the fields within, are declared as optional. ksqlDB does not yet have null constraints, meaning that today, any value can be null. To cope with this, all schemas and field values must be marked as optional.
+
+- Explictly declaring schemas is *only* needed when using structs. But because this example makes use of structs in all possible places (input, intermediate, and output values), schemas are declared for all of them.
```

Comment:
```suggestion
- Explicitly declaring schemas is *only* needed when using structs and decimals. But because this example makes use of structs in all possible places (input, intermediate, and output values), schemas are declared for all of them.
```